src/main/java/org/broadinstitute/consent/http/service/DatabaseElectionAPI.java

```diff
@@ -611,11 +611,21 @@ private void updateAccessElection(Integer electionId, String type, String status
 
     private void updateFinalVote(Election rec, Integer electionId) {
         if (rec.getFinalVote() != null) {
-            Vote vote = voteDAO.findVoteByElectionIdAndType(electionId, VoteType.CHAIRPERSON.getValue());
-            vote.setVote(rec.getFinalVote());
-            vote.setCreateDate(rec.getFinalVoteDate());
-            vote.setRationale(rec.getFinalRationale());
-            voteDAO.updateVote(vote.getVote(), vote.getRationale(), vote.getUpdateDate(), vote.getVoteId(), vote.getIsReminderSent(), vote.getElectionId(), vote.getCreateDate(), vote.getHasConcerns());
+            List<Vote> finalVotes = voteDAO.findVotesByElectionIdAndType(electionId, VoteType.FINAL.getValue());
+            Date finalVoteDate = rec.getFinalVoteDate();
+            if(finalVoteDate == null) {
+                finalVoteDate = new Date();
+            }
+            for (Vote vote : finalVotes) {
+                vote.setVote(rec.getFinalVote());
+                vote.setUpdateDate(finalVoteDate);
+                vote.setRationale(rec.getFinalRationale());
+                voteDAO.updateVote(
+                    vote.getVote(), vote.getRationale(), vote.getUpdateDate(), vote.getVoteId(),
+                    vote.getIsReminderSent(), vote.getElectionId(), vote.getCreateDate(),
+                    vote.getHasConcerns()
+                );
+            }
```

Comment:
In the context of where this is called from, `updateElectionById`, I don't think we should be modifying any votes at all. Vote modification should only ever be done when creating an election (and creating votes for that election), or when updating a vote's value directly through the Vote APIs. My recommendation here is to remove this call altogether. It is named one thing (update final vote) and then queries for a Chairperson vote, making this even more suspect.